ipv8/attestation/trustchain/payload.py

```diff
@@ -237,16 +264,17 @@ class HalfBlockPairBroadcastPayload(HalfBlockPairPayload):
     Payload for a broadcast message that ships two half blocks
     """
 
-    format_list = ['74s', 'I', '74s', 'I', '32s', '64s', 'varlenI'] * 2 + ['I']
+    format_list = ['74s', 'I', '74s', 'I', '32s', '64s', 'varlenI', 'varlenI', 'Q'] * 2 + ['I']
 
     def __init__(self, public_key1, sequence_number1, link_public_key1, link_sequence_number1, previous_hash1,
-                 signature1, transaction1, public_key2, sequence_number2, link_public_key2, link_sequence_number2,
-                 previous_hash2, signature2, transaction2, ttl):
+                 signature1, block_type1, transaction1, timestamp1, public_key2, sequence_number2, link_public_key2,
+                 link_sequence_number2, previous_hash2, signature2, block_type2, transaction2, timestamp2, ttl):
         super(HalfBlockPairBroadcastPayload, self).__init__(public_key1, sequence_number1, link_public_key1,
                                                             link_sequence_number1, previous_hash1, signature1,
-                                                            transaction1, public_key2, sequence_number2,
-                                                            link_public_key2, link_sequence_number2, previous_hash2,
-                                                            signature2, transaction2)
+                                                            block_type1, transaction1, timestamp1, public_key2,
+                                                            sequence_number2, link_public_key2, link_sequence_number2,
+                                                            previous_hash2, signature2, block_type2, transaction2,
+                                                            timestamp2)
```

Comment:
Using these many unnamed arguments with basically no formatting is asking for troubles...

Reply:
I agree. Payloads which contain much data have many arguments. Maybe we can put each argument on a new line?

Reply:
The problem is not in the formatting, but in the number of arguments and potential pitfall of mis-assigning them. I would prefer an intermediate object, or dict, or named arguments instead.
If you think it's OK to leave it as it is, that's up to you. Its only a minor note on style, nothing serious.